visualization/app/codeCharta/util/gameObjectsParser/gameObjectsImporter.ts

```diff
@@ -0,0 +1,136 @@
+		}
+	}
+
+	ccJson.data.nodes = nodes
+	ccJson.data.edges = cycles.map(cycle => createEdge(cycle))
+	ccJson.data.attributeTypes = createAttributeTypes()
+	ccJson.checksum = md5(JSON.stringify(ccJson.data))
+	return ccJson
+}
+
+function addNodeRecursively(nodeNames: string[], nodes: CodeMapNode[], parentNodeName: string, gameObjectPosition, gameObjectPositions) {
+	if (nodeNames.length === 0) return
+
+	// get current node name and create cc node structure
+	const [nodeName] = nodeNames
+	let node: CodeMapNode = {
+		name: nodeName,
+		type: isFile(nodeNames) ? NodeType.FILE : NodeType.FOLDER,
+		attributes: {}
+	}
+	if (!isFile(nodeNames)) node.children = []
+	if (isFile(nodeNames)) node = wrapFileInAFolder(nodeName, node, gameObjectPosition)
+
+	// add or select node if found
+	if (!nodeAlreadyExists(nodes, nodeName)) {
+		nodes.push(node)
+	} else {
+		node = nodes.find(node => node.name === nodeName)
+	}
+
+	const parent = gameObjectPositions.find(gameObject => gameObject.name === parentNodeName)
+	node.fixedPosition = calculateFixedFolderPosition(node, nodeName, parent, gameObjectPosition)
+
+	const newParentName = parentNodeName === "base" ? node.name : `${parentNodeName}.${node.name}`
+	addNodeRecursively(nodeNames.slice(1), node.children, newParentName, gameObjectPosition, gameObjectPositions)
+}
+
+function isFile(names: string[]) {
+	return names.length === 1
+}
+
+function nodeAlreadyExists(nodes: CodeMapNode[], name: string) {
+	return nodes.some(node => node.name === name)
+}
+
+function wrapFileInAFolder(nodeName: string, node: CodeMapNode, gameObjectPosition) {
+	const childNode = { ...node }
+	childNode.attributes = { height: gameObjectPosition.scale.y }
+	const parentNode: CodeMapNode = { name: nodeName, type: NodeType.FOLDER, attributes: {}, children: [childNode] }
+	return parentNode
+}
+
+function calculateFixedFolderPosition(node: CodeMapNode, name: string, parentGameObject, childGameObject) {
+	let position: FixedPosition
+	if (node.type === NodeType.FOLDER && name !== "base") {
+		const cornerXofParent = parentGameObject.position.x - parentGameObject.scale.x / 2
+		const cornerZofParent = parentGameObject.position.z - parentGameObject.scale.z / 2
+		const cornerXofChild = childGameObject.position.x - childGameObject.scale.x / 2
+		const cornerZofChild = childGameObject.position.z - childGameObject.scale.z / 2
+		const top = Math.floor(((cornerXofChild - cornerXofParent) / parentGameObject.scale.x) * 100)
+		const left = Math.floor(((cornerZofChild - cornerZofParent) / parentGameObject.scale.z) * 100)
+		const width = Math.floor((childGameObject.scale.z / parentGameObject.scale.z) * 100)
+		const height = Math.floor((childGameObject.scale.x / parentGameObject.scale.x) * 100)
+		position = { left, top, width, height }
+	}
+	if (name === "root") {
+		position.top = Math.floor(50 - position.height / 2)
+		position.left = 0
+	}
+	return position
+}
```

Comment:
Let's add some short explanation why the division by 2 is necessary and why position.top is calculated as it is.
It's hard to remember the reasons for those calculations.

Reply:
I thought the variable name is descriptive enough. But you are right better have a more detailed description in this case. 
Will add a few comments 👍